Cauli/Cauli.swift

```diff
@@ -124,8 +124,9 @@ extension Cauli: CauliURLProtocolDelegate {
                 completion(record)
             }
         }, completion: { record in
+            let modifiedRecord = self.configuration.prePersistHook?(record) ?? record
```

Comment:
I am currently thinking where the better place to apply this modifier-closure would be. Here or within the storage itself.

* If we do it here, we run the risk of forgetting it if we ever want to store a record from a different line.
* If we do it in the storage itself every storage (if we ever actually can replace the storage) needs to actually apply the modifier.

What do you like better?

Reply:
I like it better to put it in the Storage implementation because it's a more central place. But I am at a loss on how to "enforce" future implementations to call the hook at the right time. The only thing that comes to mind is really bold documentation.

Reply:
Hm ... I am a little unsure about that. What do you think @pstued ?

Reply:
In this case I would go with the storage as well.
I believe whenever we introduce a replaceable storage, that will be a major change. So we can otherthink if a `prePersistModifier` is still needed in the general Storage protocol or should be moved to a dedicated storage.

Reply:
On the other side, the need to modify a record before storing it, feels like a use case for a dedicated storage. In that case, it could make sense to allow the initialisation of Cauli with a custom Storage fulfilling that need.

Reply:
For now I always perceived the storage as defining how to store something, not what to store. This is why I would argue for not using a specific storage that is different to any other storage in what it stored.

I guess for now I would probably move `prePersistModifier` into the one storage we have and change it as new requirements appear.

Reply:
If @pstued is dine with this I will go ahead and commit the change.

Reply:
go for it